src/main/java/group1/mips_simulator/components/cpuParts/RomLoader.java

```diff
@@ -1,20 +1,33 @@
 package group1.mips_simulator.components.cpuParts;
 
+import group1.mips_simulator.Utility;
+import group1.mips_simulator.components.ROM;
+import group1.mips_simulator.components.instructionParts.Instruction;
+import group1.mips_simulator.components.memParts.Memory;
+
+import group1.mips_simulator.components.instructionParts.Instruction;
+import java.util.Vector;
+
 public class RomLoader {
+    Memory memory;
```

Comment:
It's a little weird to me that a `RomLoader` has-a `Memory`. I'm assuming this is a reference to the same memory that the `Computer` has?

Reply:
yea, it is, though i agree its pretty weird for the romloader to be dedicated to a memory, so ill change that